### ISSUE TITLE
Local backend file format should be architecture independent

### ISSUE THREAD

**Issue by SCDealy** [s3ql/s3ql#41]
I'm currently copying a large S3QL file system (6 TB raw data, 800 GB deduplicated/compressed) from a 32 bit x86 system to a 64 bit ARM, both using the "local" backend. Initially, I naively simply copied the S3QL directory tree between the two systems, but the ARM system was unable to open the copied file system. After a bit of investigation, I realized that some aspects of common Python binary file formats are architecture dependent.

This may not be practical to implement, and presumably depends on whether the format of the compressed data blocks are processor architecture independent (some compression formats are). It also may not be worth doing as it is presumably a rare use-case, however, if you don't ask ... :-)

I am currently transferring the file system the hard way (fortunately, they are on a local network so CPU speed on the receiving end is the primary limiting factor). I estimate this will take a few weeks to a few months depending on how performance changes as the received data shifts from new data to primarily duplicate data.

**Comment by Nikratio** [comment]
Thanks for the report! The format is intended to be architecture independent. Could you describe the differences that you found?

**Comment by SCDealy** [comment]
On Sun, 21 Oct 2018, Nikolaus Rath wrote:

> Thanks for the report! The format is intended to be architecture 
> independent. Could you describe the differences that you found?

Apparently I am getting senile. I thought the x86 box was running stretch and 
therefore the same version of S3QL as the ARM system. The problem is not 
that they are incompatible due to architecture, but due to S3QL version 
differences. Since I knew the file system wasn't corrupt and thought they 
were running the same software, I interpreted the crash as an architecture 
dependency problem.

I have opened a different issue since it would be friendlier if it reported 
the problem and didn't throw an exception when attempting to mount an 
incompatible file system version.

Shannon C. Dealy               |       DeaTech Research Inc.
dealy@deatech.com              | Biotechnology Development Services
Telephone USA: +1 541-929-4089 |      USA and the Netherlands
Netherlands:   +31 85 208 5570 |          www.deatech.com
